### DISCUSSION TITLE
404 SSG pages are not revalidated on next 11.1.2

### DISCUSSION THREAD

**Discussion by aomini** [vercel/next.js#28688] [Help] [answered]
### What version of Next.js are you using?

11.1.2

### What version of Node.js are you using?

14.17.4

### What browser are you using?

chrome

### What operating system are you using?

ubuntu

### How are you deploying your application?

next start and vercel as well

### Describe the Bug

I have a page that catch all routes on `/products/[slug].jsx`. I have implemented ISR and `fallback: true`. On `getStaticPaths`, I'm fetching the products that are `published`.

The bug I'm facing is I have a published product and to make some changes I changed it's status to unpublished which in return visiting to `/products/my-product` returns `404`. Then, after I'm done with the changes and changed the status to published it keeps showing me the 404 page on reload. But if I visit the product from `<Link />` then it works as expected.

### Expected Behavior

A page should be able to revalidate itself from 404.

### To Reproduce

Here's my `getStaticProps` function:
```jsx
export const getStaticProps = async ({ params }) => {
  const { slug } = params;

  const data = await fetch(`http://mydomaim/find/${slug}`);

  if (data.status === 404) {
    return {
      notFound: true,
    };
  }
  const product = await data.json();

  return {
    props: {
      name: product.name,
    },
    revalidate: 10,
  };
};
```

**Comment by ijjk** [comment, marked as answer]
Hi, `revalidate` needs to be returned when returning `notFound: true` as well for the page to continue to revalidate otherwise the default `revalidate` value of `false` is used. 

Related documentation here https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation

**Reply by aomini** [reply]
Great help thanks

**Reply by trm217** [reply]
In that case I believe the TS docs for the GetStaticPropsResult type is wrong:
```ts
export type GetStaticPropsResult<P> =
  | { props: P; revalidate?: number | boolean }
  | { redirect: Redirect; revalidate?: number | boolean }
  | { notFound: true }
  ```
  As seen above, when returning a notFound-result, no revalidate value is expected to be set.

**Reply by aomini** [reply]
yes but I believe it made it working.
